### DISCUSSION TITLE
Working with arrays

### DISCUSSION THREAD

**Discussion by zyberpunker** [msiemens/tinydb#441] [Q&A]
Is it possible to work with arrays in tinydb? 
For example, how can I update an array with cars if I would like to add 'bmw' to the following?
{'name': 'Bob', 'cars': ['volvo', 'mazda']}

**Comment by msiemens** [comment]
Hey @zyberpunker,

you use `db.update` and pass an update function as the second parameter (see https://tinydb.readthedocs.io/en/latest/usage.html#updating-data although it doesn't make this pattern clear):

```python
db.update(where('name') == 'Bob, lambda doc: doc['contents'].append('bmw'))
```

The `lambda` here gets the matching document and can modify it however it likes. You can append data, remove it etc.

Can you verify this works for you?

**Reply by MrLucio** [reply]
Hi,

thank you for the reply, I too was looking for a `push`-like method.
I can confirm it works with a minor change to `db.update`, regarding the order or the parameters:

```python
db.update(lambda doc: doc['cars'].append('bmw'), where('name') == 'Bob')
```

The order of parameters you used will cause the updates issued by the `lamda` to be applied to all documents with no filtering whatsoever.

**Reply by msiemens** [reply]
Oh, you're right, I confused the order of the arguments! It's the operation first, then the search condition 🙂
